src/main/scala/com/databricks/spark/sql/perf/results.scala

```diff
@@ -97,3 +102,48 @@ case class BreakdownResult(
     delta: Double)
 
 case class Failure(className: String, message: String)
+
+case class MLTestParameters(
+    numFeatures: Option[Int] = None,
+    numExamples: Option[Long] = None,
+    numTestExamples: Option[Long] = None,
+    numPartitions: Option[Int] = None,
+    randomSeed: Option[Int] = None)
+
+object MLTestParameters {
+
+  val empty = MLTestParameters()
+}
+
+// KEEP ARGUMENTS SORTED BY NAME
+case class ExtraMLTestParameters(
```

Comment:
You are right, I am going to combine both classes.

Regarding the number of fields, this issue will disappear when we move this year to scala 2.11, and we are already reusing some parameters.

I am not keen on a big map, because the typing information acts as a basic sanity check.